appgate/files.py

```diff
@@ -0,0 +1,144 @@
+import asyncio
+import base64
+import os
+
+import aiohttp
+import requests
+from typing import Optional, Dict, List
+
+import urllib3.response
+from aiohttp import ClientSession
+from minio import Minio  # type: ignore
+
+from appgate.customloaders import FileAttribLoader
+from appgate.openapi.attribmaker import AttribMaker
+from appgate.openapi.types import (
+    AttribType,
+    OpenApiDict,
+    EntityClassGeneratorConfig,
+    AttributesDict,
+    K8S_LOADERS_FIELD_NAME,
+)
+
+
+class AppgateFileException(Exception):
+    pass
+
+
+class AppgateFile:
+    """
+    AppgateFile base abstract class. Every file field in an entity will
+    be an AppgateFile instance that will be able to load its value
+    """
+
+    def __init__(self, value: Dict, entity_name: str) -> None:
+        self.value = value
+        self.entity_name = entity_name
+        self.api_version = os.getenv("APPGATE_API_VERSION")
+
+    def load_file(self) -> str:
+        raise NotImplementedError()
+
+
+class AppgateHttpFile(AppgateFile):
+    def load_file(self) -> str:
+        file_key = f"{self.entity_name.lower()}-{self.api_version}/{self.value.get('filename')}"
+        address = os.getenv("APPGATE_FILE_HTTP_ADDRESS")
+        file_url = f"{address}/{file_key}"
+
+        async def get(url: str) -> str:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url) as response:
+                    return base64.b64encode(await response.read()).decode()
+
+        return asyncio.run(get(file_url))
```

Comment:
I was thinking more of making `load_file` itself an async method since the operator already uses asyncio at the top-level anyways.

Reply:
I agree, I think it makes more sense to do the asyncio.run on the generic code, the whole operator is mostly async anyway